diesel/src/query_source/aliasing/alias.rs

```diff
@@ -0,0 +1,171 @@
+{
+    type FromClause = Self;
+    type DefaultSelection =
+        <<S::Table as QuerySource>::DefaultSelection as FieldAliasMapper<S>>::Out;
+
+    fn from_clause(&self) -> Self::FromClause {
+        self.clone()
+    }
+
+    fn default_selection(&self) -> Self::DefaultSelection {
+        self.fields(S::Table::table().default_selection())
+    }
+}
+
+impl<S, DB> QueryFragment<DB> for Alias<S>
+where
+    S: AliasSource,
+    DB: Backend,
+    S::Table: StaticQueryFragment,
+    <S::Table as StaticQueryFragment>::Component: QueryFragment<DB>,
+{
+    fn walk_ast<'b>(&'b self, mut pass: AstPass<'_, 'b, DB>) -> QueryResult<()> {
+        <S::Table as StaticQueryFragment>::STATIC_COMPONENT.walk_ast(pass.reborrow())?;
+        pass.push_sql(" AS ");
+        pass.push_identifier(S::NAME)?;
+        Ok(())
+    }
+}
+
+impl<S> AsQuery for Alias<S>
+where
+    S: AliasSource,
+    S::Table: AsQuery,
+    Self: QuerySource,
+    <Self as QuerySource>::DefaultSelection: ValidGrouping<()>,
+{
+    type SqlType = <<Self as QuerySource>::DefaultSelection as Expression>::SqlType;
+    type Query = SelectStatement<FromClause<Self>>;
+
+    fn as_query(self) -> Self::Query {
+        SelectStatement::simple(self)
+    }
+}
+
+impl<S, QS> AppearsInFromClause<QS> for Alias<S>
+where
+    S: AliasSource,
+    S::Table: AliasAppearsInFromClause<S, QS>,
+{
+    type Count = <S::Table as AliasAppearsInFromClause<S, QS>>::Count;
+}
+#[doc(hidden)]
+/// This trait is used to allow external crates to implement
+/// `AppearsInFromClause<QS> for Alias<S>`
+///
+/// without running in conflicting impl issues
+pub trait AliasAppearsInFromClause<S, QS> {
+    /// Will be passed on to the `impl AppearsInFromClause<QS>`
+    type Count;
+}
+#[doc(hidden)]
+/// This trait is used to allow external crates to implement
+/// `AppearsInFromClause<Alias<S2>> for Alias<S1>`
+///
+/// without running in conflicting impl issues
+pub trait AliasAliasAppearsInFromClause<T2, S1, S2> {
+    /// Will be passed on to the `impl AppearsInFromClause<QS>`
+    type Count;
+}
+impl<T1, S1, S2> AliasAppearsInFromClause<S1, Alias<S2>> for T1
+where
+    S2: AliasSource,
+    T1: AliasAliasAppearsInFromClause<S2::Table, S1, S2>,
+{
+    type Count = <T1 as AliasAliasAppearsInFromClause<S2::Table, S1, S2>>::Count;
+}
+
+// impl<S: AliasSource<Table=T1>> AppearsInFromClause<T2> for Alias<S>
+// where T1 != T2
+impl<T1, T2, S> AliasAppearsInFromClause<S, T2> for T1
+where
+    T1: TableNotEqual<T2> + Table,
+    T2: Table,
+    S: AliasSource<Table = T1>,
+{
+    type Count = Never;
+}
+
+// impl<S1, S2> AppearsInFromClause<Alias<S1>> for Alias<S2>
+// where S1: AliasSource, S2: AliasSource, S1::Table != S2::Table
+impl<T1, T2, S1, S2> AliasAliasAppearsInFromClause<T1, S2, S1> for T2
+where
+    T1: TableNotEqual<T2> + Table,
+    T2: Table,
+    S1: AliasSource<Table = T1>,
+    S2: AliasSource<Table = T2>,
+{
+    type Count = Never;
+}
+
+impl<S> AppearsInFromClause<Alias<S>> for ()
```

Comment:
This impl shoud not be needed anymore, as we don't allow `()` as query source anymore.